torch/testing/_internal/common_distributed.py

```diff
@@ -111,6 +115,8 @@ def wrapper(*args, **kwargs):
         world_size = int(os.environ["WORLD_SIZE"])
         if torch.cuda.device_count() < world_size:
             sys.exit(TEST_SKIPS[f"multi-gpu-{world_size}"].exit_code)
+        if TEST_HPU and torch.hpu.device_count < world_size:
+            sys.exit(TEST_SKIPS[f"multi-gpu-{world_size}"].exit_code)
```

Comment:
Can we prepare a map and unify all the checks into one?
```
if DEVICE_COUNT[DEVICE_UNDER_TEST] < world_size:
```

Reply:
Thank you for your suggestion. It's an interesting idea to unify the checks into a single map. While this approach could potentially improve readability, I have a few concerns:

1. While a globally available map could potentially unify these checks, I'm uncertain about the feasibility of creating such a map that would be accessible to all functions within common distributed. 

2. This map would then have to be created independently for each function that uses the device count. This approach could lead to two potential issues:

	a. Increased memory usage: Creating multiple instances of the same map across different functions could lead 	to unnecessary memory overhead.

	b. Processing overhead: We would still need to follow similar conditional logic each time the map is created 	or accessed. This means we might not be reducing complexity or processing time, but rather replacing one set 	of conditionals with another.

Given these considerations, I'm not convinced that the benefits of this approach would outweigh the potential drawbacks.

I would love to get your perspective on this. Have I overlooked any potential benefits or alternative implementations that could address these concerns?